type mismatch (long vs float) in CAffinityPropagation

### Expected behaviour
No compiler warning about type mismatch in CAffinityPropagation function declaration and definition.

### Actual behaviour
```
[...]
gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m32 -march=i686 -fasynchronous-unwind-tables -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m32 -march=i686 -fasynchronous-unwind-tables -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m32 -march=i686 -fasynchronous-unwind-tables -fPIC -I/usr/lib/python2.7/site-packages/numpy/core/include -IMDAnalysis/analysis/encore/clustering/include -I/usr/include/python2.7 -c MDAnalysis/analysis/encore/clustering/affinityprop.c -o build/temp.linux-i686-2.7/MDAnalysis/analysis/encore/clustering/affinityprop.o -O3 -ffast-math -std=c99
In file included from /usr/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1809:0,
                 from /usr/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                 from /usr/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from MDAnalysis/analysis/encore/clustering/affinityprop.c:460:
/usr/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprec
ated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^~~~~~~
MDAnalysis/analysis/encore/clustering/affinityprop.c: In function '__pyx_pf_10MDAnalysis_8analysis_6encore_10clustering_12affinityprop_19AffinityPropagation_run':
MDAnalysis/analysis/encore/clustering/affinityprop.c:2221:180: warning: passing argument 7 of 'CAffinityPropagation' from incompatible pointer type [-Wincompatible-pointer-types]
   __pyx_t_10 = __Pyx_PyInt_From_int(CAffinityPropagation(((float *)__pyx_v_matndarray->data), __pyx_v_cn, __pyx_v_lam, __pyx_v_max_iterations, __pyx_v_convergence, __pyx_v_noise, ((long *)__pyx_v_clusters->data))); if (unlikely(!__pyx_t_10)) __PYX_ERR(0, 104, __pyx_L1_error)
                                                                                                                                                                                    ^
In file included from MDAnalysis/analysis/encore/clustering/affinityprop.c:465:0:
MDAnalysis/analysis/encore/clustering/include/ap.h:30:5: note: expected 'float *' but argument is of type 'long int *'
 int CAffinityPropagation(float*, int, float, int, int, int, float*);
     ^~~~~~~~~~~~~~~~~~~~
```

### Code to reproduce the behaviour

``` bash
$ grep -n CAffinityPropagation MDAnalysis/analysis/encore/clustering/{caffinityprop.pxd,include/ap.h} 
MDAnalysis/analysis/encore/clustering/caffinityprop.pxd:42:    int CAffinityPropagation(float*, int, float, int, int, bint, long*)
MDAnalysis/analysis/encore/clustering/include/ap.h:30:int CAffinityPropagation(float*, int, float, int, int, int, float*);
```

I'm not sure which type is correct here, but it seems it's only ever called from `affinityprop.pyx`
and the parameter is converted to `long*` anyway:
```
$ grep -n CAffinityPropagation MDAnalysis/analysis/encore/clustering/affinityprop.pyx
MDAnalysis/analysis/encore/clustering/affinityprop.pyx:104:        iterations = caffinityprop.CAffinityPropagation( <float*>matndarray.data, cn, lam, max_iterations, convergence, noise, <long*>clusters.data)
```

### Currently version of MDAnalysis:
0.16.1

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -13,7 +13,7 @@ The rules for this file:
   * release numbers follow "Semantic Versioning" http://semver.org
 
 ------------------------------------------------------------------------------
-mm/dd/17 richardjgowers, rathann, orbeckst, tylerjereddy
+mm/dd/17 richardjgowers, rathann, orbeckst, tylerjereddy, mtiberti
 
   * 0.17.0
 
@@ -24,6 +24,8 @@ Deprecations
 Fixes
   * correctly read little-endian TRZ files on big-endian architectures (issue
     #1424)
+  * Fixed type matching and inclusion compilation warnings for the
+  ENCORE analysis package (issue #1390)
 
 Changes
   * remove deprecated TimeSeriesCollection

--- a/package/MDAnalysis/analysis/encore/clustering/include/ap.h
+++ b/package/MDAnalysis/analysis/encore/clustering/include/ap.h
@@ -27,4 +27,4 @@ float min(float*, int);
 
 float max(float*, int);
 
-int CAffinityPropagation(float*, int, float, int, int, int, float*);
+int CAffinityPropagation(float*, int, float, int, int, int, long*);

--- a/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c
+++ b/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c
@@ -19,13 +19,14 @@
   MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
   J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
 */
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-//#include <math.h>
+#include <math.h>
 #include <time.h>
 #include <sys/types.h>
-#include <time.h>
+#include <unistd.h>
 
 #define EPSILON 1e-8
 
@@ -179,7 +180,7 @@ double CStochasticProximityEmbedding(
         int nstep,
         int stressfreq) {
 
-    int a = 0, b = 0, idx = 0, idxa = 0, idxb = 0, idxak = 0, idxbk = 0;
+    int a = 0, b = 0, idxa = 0, idxb = 0, idxak = 0, idxbk = 0;
     double dab = 0.0, rab = 0.0;
     double lam = maxlam;
     double t = 0.0;

